test/util/grpctest/mocks.go

```diff
@@ -0,0 +1,108 @@
+package grpctest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"sync/atomic"
+	"testing"
+
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// MockTxService allows controlling the behavior of BroadcastTx calls.
+type MockTxService struct {
+	sdktx.UnimplementedServiceServer // Embed the unimplemented server
+
+	BroadcastHandler func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error)
+	Invocations      atomic.Int32 // Use atomic for potential concurrent access if needed
+}
+
+func (m *MockTxService) BroadcastTx(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+	m.Invocations.Add(1)
```

Comment:
are you using this for something?

Reply:
yeah, it's used in broadcast test as a mock endpoint

Reply:
I meant specifically tracking the Invocations -- where do you use this?